sky/provision/vast/utils.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=assignment-from-no-return
+"""Vast library wrapper for SkyPilot."""
+from typing import Any, Dict, List, Optional
+
+from sky import sky_logging
+from sky.adaptors import vast
+
+logger = sky_logging.init_logger(__name__)
+
+
+def list_instances() -> Dict[str, Dict[str, Any]]:
+    """Lists instances associated with API key."""
+    instances = vast.vast().show_instances()
+
+    instance_dict: Dict[str, Dict[str, Any]] = {}
+    for instance in instances:
+        instance['id'] = str(instance['id'])
+        info = instance.copy()
+
+        if isinstance(instance['actual_status'], str):
+            info['status'] = instance['actual_status'].upper()
+        else:
+            info['status'] = 'UNKNOWN'
+        info['name'] = instance['label']
+
+        instance_dict[instance['id']] = info
+
+    return instance_dict
+
+
+def launch(name: str, instance_type: str, region: str, disk_size: int,
+           image_name: str, ports: Optional[List[int]], public_key: str) -> str:
+    """Launches an instance with the given parameters.
+
+    Converts the instance_type to the Vast GPU name, finds the specs for the
+    GPU, and launches the instance.
+    """
+    del ports
+    del public_key
+
+    gpu_name = instance_type.split('-')[1].replace('_', ' ')
+    num_gpus = int(instance_type.split('-')[0].replace('x', ''))
+
+    query = ' '.join([
+        f'geolocation="{region[-2:]}"',
```

Comment:
What is this `-2` for ?

Reply:
Geolocation on vast can be as specific as the user chooses to be. They can say, for instance, "Yutakacho, Shinagawa District, Tokyo, JP."  This is different from most providers who use the 2-letter country code such as JP and such a geolocation would fail to return results in a simple string comparison. 

Regardless of specificity, all our geolocations end in two-letter country codes so we just snip that to conform to how many providers state their location.

Reply:
> Regardless of specificity, all our geolocations end in two-letter country codes so we just snip that to conform to how many providers state their location.

can we add this to the comment?

Reply:
see a9d2ff451ddd0b7e24c2c49c6e8467f43a218344